src/Core/src/Credentials/ConfigurationProvider.php

```diff
@@ -29,4 +51,37 @@ public function getCredentials(Configuration $configuration): ?Credentials
 
         return null;
     }
+
+    private function getCredentialsFromRole(string $roleArn, ?string $sessionName, ?string $region): ?Credentials
+    {
+        $sessionName = $sessionName ?? \uniqid('async-aws-', true);
+
+        $stsClient = new StsClient(['region' => $region], new NullProvider(), $this->httpClient);
```

Comment:
I think `NullProvider` wont work. You need a key/secret somewhere.

It should be something similar to what have been generated in `getCredentials` method
```php
new Credentials(
  $configuration->get(Configuration::OPTION_ACCESS_KEY_ID),
  $configuration->get(Configuration::OPTION_SECRET_ACCESS_KEY),
  $configuration->get(Configuration::OPTION_SESSION_TOKEN)
);
```